### ISSUE TITLE
requests.exceptions.HTTPError: 429 Client Error: Too Many Requests for url: https://www.pinterest.com/resource/UserSessionResource/create/

### ISSUE THREAD

**Issue by thaeralmadhoun** [bstoilov/py3-pinterest#124]
Thank you for creating this great package. I used this package before to pin many images to my pinterest account. But now I'm facing an issue. I'm trying to use the package again and to login I'm using pinterest.login() but I get this error:

Traceback (most recent call last):
  File "C:/Users/thn87/PycharmProjects/pinDC/pinDC.py", line 16, in <module>
    pinterest.login()
  File "C:\Program Files\Python37\lib\site-packages\py3pin\Pinterest.py", line 131, in login
    return self.post(url=CREATE_USER_SESSION, data=data)
  File "C:\Program Files\Python37\lib\site-packages\py3pin\Pinterest.py", line 111, in post
    return self.request('POST', url=url, data=data, files=files, extra_headers=headers)
  File "C:\Program Files\Python37\lib\site-packages\py3pin\Pinterest.py", line 103, in request
    response.raise_for_status()
  File "C:\Program Files\Python37\lib\site-packages\requests\models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 429 Client Error: Too Many Requests for url: https://www.pinterest.com/resource/UserSessionResource/create/

**Comment by bstoilov** [comment]
You have called pinterest.login() too many times.

You are not calling it every time before you pin right?

**Comment by ivansaul** [comment]
Hi @thaeralmadhoun, I also got the same error. Could you solve it?
I had already done a couple of tests before and everything was going well, but from a moment ago I got the error. @bstoilov    any idea, how I could solve it? 
This is my code:
```python
from py3pin.Pinterest import Pinterest
import pandas as pd
import time

email='email@1adir.com'
password='pass'
username='username'
pinterest = Pinterest(email=email,
                      password=password,
                      username=username,
                      cred_root='cred_root')
#pinterest.login(headless=False)

df=pd.read_excel('data.xlsx')
for index,row in df.iterrows():
    pinterest.pin(board_id='5789898654654655',
                section_id='565656566584',
                image_url=f"https://gitlab.com/{row['path']}/-/raw/master/covers/{row['name']}.png",
                description=f"🔶{row['description']}",
                title=f"{row['title']}",
                link='https://www.google.com')
    time.sleep(5)
```
**Output:**
```bash
HTTPError                                 Traceback (most recent call last)
<ipython-input-66-faac5de50c06> in <module>
     11 for index,row in df.iterrows():
---> 12     pinterest.pin(board_id='5789898654654655',
     13                 section_id='565656566584',
     14                 image_url=f"https://gitlab.com/{row['path']}/-/raw/master/covers/{row['name']}.png",

~/Documentos/python3.9/site-packages/py3pin/Pinterest.py in pin(self, board_id, image_url, description, link, title, section_id)
    439         data = self.req_builder.buildPost(options=options, source_url=source_url)
    440 
--> 441         return self.post(url=PIN_RESOURCE_CREATE, data=data)
    442 
    443     def upload_pin(self, board_id, image_file, description='', link='', title='', section_id=None):

~/Documentos/python3.9/site-packages/py3pin/Pinterest.py in post(self, url, data, files, headers)
    113 
    114     def post(self, url, data=None, files=None, headers=None):
--> 115         return self.request('POST', url=url, data=data, files=files, extra_headers=headers)
    116 
    117     def login(self, headless=True, wait_time=15, proxy=None):

~/Documentos/python3.9/site-packages/py3pin/Pinterest.py in request(self, method, url, data, files, extra_headers)
    105 
    106         response = self.http.request(method, url, data=data, headers=headers, files=files, proxies=self.proxies)
--> 107         response.raise_for_status()
    108 
    109         return response

~/Documentos/python3.9/site-packages/requests/models.py in raise_for_status(self)
    941 
    942         if http_error_msg:
--> 943             raise HTTPError(http_error_msg, response=self)
    944 
    945     def close(self):

HTTPError: 429 Client Error: Too Many Requests for url: https://www.pinterest.com/resource/PinResource/create/
```

**Comment by thaeralmadhoun** [comment]
> You have called pinterest.login() too many times.
> 
> You are not calling it every time before you pin right?

No. I call it only once. I waited for a couple of days to see if it had something to do with Pinterest rate-limiting, however, that didn't solve the issue.

**Comment by thaeralmadhoun** [comment]
> Hi @thaeralmadhoun, I also got the same error. Could you solve it?
> I had already done a couple of tests before and everything was going well, but from a moment ago I got the error. @bstoilov any idea, how I could solve it?
> This is my code:
> 
> ```python
> from py3pin.Pinterest import Pinterest
> import pandas as pd
> import time
> 
> email='email@1adir.com'
> password='pass'
> username='username'
> pinterest = Pinterest(email=email,
>                       password=password,
>                       username=username,
>                       cred_root='cred_root')
> #pinterest.login(headless=False)
> 
> df=pd.read_excel('data.xlsx')
> for index,row in df.iterrows():
>     pinterest.pin(board_id='5789898654654655',
>                 section_id='565656566584',
>                 image_url=f"https://gitlab.com/{row['path']}/-/raw/master/covers/{row['name']}.png",
>                 description=f"🔶{row['description']}",
>                 title=f"{row['title']}",
>                 link='https://www.google.com')
>     time.sleep(5)
> ```
> 
> **Output:**
> 
> ```shell
> HTTPError                                 Traceback (most recent call last)
> <ipython-input-66-faac5de50c06> in <module>
>      11 for index,row in df.iterrows():
> ---> 12     pinterest.pin(board_id='5789898654654655',
>      13                 section_id='565656566584',
>      14                 image_url=f"https://gitlab.com/{row['path']}/-/raw/master/covers/{row['name']}.png",
> 
> ~/Documentos/python3.9/site-packages/py3pin/Pinterest.py in pin(self, board_id, image_url, description, link, title, section_id)
>     439         data = self.req_builder.buildPost(options=options, source_url=source_url)
>     440 
> --> 441         return self.post(url=PIN_RESOURCE_CREATE, data=data)
>     442 
>     443     def upload_pin(self, board_id, image_file, description='', link='', title='', section_id=None):
> 
> ~/Documentos/python3.9/site-packages/py3pin/Pinterest.py in post(self, url, data, files, headers)
>     113 
>     114     def post(self, url, data=None, files=None, headers=None):
> --> 115         return self.request('POST', url=url, data=data, files=files, extra_headers=headers)
>     116 
>     117     def login(self, headless=True, wait_time=15, proxy=None):
> 
> ~/Documentos/python3.9/site-packages/py3pin/Pinterest.py in request(self, method, url, data, files, extra_headers)
>     105 
>     106         response = self.http.request(method, url, data=data, headers=headers, files=files, proxies=self.proxies)
> --> 107         response.raise_for_status()
>     108 
>     109         return response
> 
> ~/Documentos/python3.9/site-packages/requests/models.py in raise_for_status(self)
>     941 
>     942         if http_error_msg:
> --> 943             raise HTTPError(http_error_msg, response=self)
>     944 
>     945     def close(self):
> 
> HTTPError: 429 Client Error: Too Many Requests for url: https://www.pinterest.com/resource/PinResource/create/
> ```

No I didn't find a solution yet. Waiting for the creator's reply.

**Comment by bstoilov** [comment]
You probably got suspended by pinterest. 

https://help.pinterest.com/en-gb/article/rate-limit-blocks

The only limit I was able to find is
_1000 calls per endpoint per hour for each unique user token_

Don't pin more than 25 times a day. New accounts are under more strict investigation, once your account has 10k+ followers and is at least 6 month old you can increase that number. With my main account I have 35k followers I always pinned every 10 minutes in 14-16 hour period. The account is 5 years old.

Try to do various things not just pin, dont only one thing is very suspicious

And finally try to pin through the UI and see if you are blocked there as well
